File: apps/benchmarks/micro/04_function_call.py
#!/usr/bin/env python3
# ベンチマーク4: 関数呼び出しオーバーヘッド (Python版)
# 期待値: 1000

def add(a, b):
    return a + b

def main():
    sum = 0
    for i in range(1000):
        sum = add(sum, 1)
    print(f"Result: {sum}")
    return 0

if __name__ == "__main__":
    main()

File: apps/benchmarks/micro/04_function_call.c
// ベンチマーク4: 関数呼び出しオーバーヘッド (C言語版)
// 期待値: 1000

#include <stdio.h>

int add(int a, int b) {
    return a + b;
}

int main() {
    int sum = 0;
    int i = 0;

    while (i < 1000) {
        sum = add(sum, 1);
        i = i + 1;
    }

    printf("Result: %d\n", sum);
    return 0;
}
